### PR TITLE
Added link rewriting, DocumentContext

### PR DESCRIPTION
**Pull request by jbheard**
Related to #11

### CODE REVIEW

File: src/Render/Markdown/DocumentContext.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace D2L.Dev.Docs.Render.Markdown {
+	// TODO: Add more properties for context
+	// Some possibilities could be github repo url, list of docs repos
+	internal sealed class DocumentContext {
```
Comment (jbheard) [inline src/Render/Markdown/DocumentContext.cs]: This is meant to be a way for us to provide metadata about the document being rendered to our custom renderers. 
Currently it just stores the name of the repository the doc lives in, but as we need things they can be added here.

Comment (j3parker) [inline src/Render/Markdown/DocumentContext.cs]: Nice

File: src/Render/Program.cs [hunk 2]
```diff
@@ -48,9 +48,10 @@ internal static class Program {
 
 			var text = await File.ReadAllTextAsync( filename );
 
+			var context = new DocumentContext( "test" );
```
Comment (jbheard) [inline src/Render/Program.cs]: How will we get the name of the repo for this? Should we add an argument to the program for which docroot we are working in?

Comment (j3parker) [reply src/Render/Program.cs]: I think that sounds reasonable. Your entrypoint can probably get it from an env var? Check the GitHub Actions docs.... there should be something there.

Comment (j3parker) [reply src/Render/Program.cs]: https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables

**Pull request merged**